### PR TITLE
perf(rust, python): use a trusted anyvalue builder

### PR DESCRIPTION
**Pull request by ritchie46**
While rewriting streaming groupby, I needed an `AnyValueBuilder`. The current one we use can handle a lot of dynamic untrusted data, which is great for dealing with user data, but is much  more expensive than it needs to be when we feed it values we know are correct.

This creates a builder for such data. This PR let's `transpose` use that builder already, so it can have a small speedup when dealing with logical data transposes and/or utf8 data.

Next logical candidate for this is the `pivot` operation.

### CODE REVIEW

File: py-polars/tests/unit/operations/test_transpose.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    assert_frame_equal(expected, out)
+
+
+def test_transpose_logical_data() -> None:
+    assert pl.DataFrame(
+        {
+            "a": [date(2022, 2, 1), date(2022, 2, 2), date(2022, 1, 3)],
+            "b": [datetime(2022, 1, 1), datetime(2022, 1, 2), datetime(2022, 1, 3)],
+        }
+    ).transpose().to_dict(False) == {
```
Comment (ritchie46) [inline py-polars/tests/unit/operations/test_transpose.py:109]: Will fix this. :)

**Comment by ghuls** [comment]
Can you also use `assert_frame_equal` in the tests?

**Comment by ritchie46** [comment]
> Can you also use `assert_frame_equal` in the tests?

It could, but why?

**Comment by ghuls** [comment]
> It could, but why?

I thought the plan was for test to use this more by default, as it also allows to check for correct dtypes easily.

**Comment by ritchie46** [comment]
> I thought the plan was for test to use this more by default, as it also allows to check for correct dtypes easily.

I think a bit of variance on the assert methods leads to extra coverage. Moving data to python dictionaries has revealed a lot of bugs in coversion of anyvalues. I think both are fine to use interchangebly.

**Pull request merged**